### PR TITLE
Add "Open Video…" menu item to more menu

### LINKED ISSUE #568: Add a `Open Video…` menu item
Add a `Open Video…` menu item in the `...` menu in the cropper controls view. To make it clear that we also support importing existing videos. Users can already drag and drop a video into the menu bar icon, but that's not obvious.

### PR DESCRIPTION
**Pull request by theshortcut**
Fixes #568 

Adds an "Open Video…" menu item which creates an openFile type dialog, which then opens selected videos in the editor.

oct-06-2018 23-41-35

### CODE REVIEW

**Comment by skllcrn** [comment]
This is a really good idea @theshortcut, great first-time contributions, we're glad to have you!

File: main/menus.js [hunk 1]
```diff
@@ -40,6 +42,24 @@ const cogMenuTemplate = [
...
     type: 'separator'
   },
+  {
+    label: 'Open Video…',
+    accelerator: 'Cmd+o',
+    click: () => {
+      closeAllCroppers();
+      dialog.showOpenDialog({
+        filters: [
+          {name: 'Videos', extensions: ['mov', 'mp4', 'm4v']},
```
Comment (sindresorhus) [inline main/menus.js]: We already define the extensions here: https://github.com/wulkano/kap/blob/0cbff512718ed3f31bdbc0913364df57deb8d2fe/main/tray.js#L12 We should put it in a common place to we only have to update it in one place. Maybe a `/main/common/constants.js` file? You'll have to add the dot here instead: https://github.com/wulkano/kap/blob/0cbff512718ed3f31bdbc0913364df57deb8d2fe/main/tray.js#L25

File: main/menus.js [hunk 2]
```diff
@@ -40,6 +42,24 @@ const cogMenuTemplate = [
...
+          {name: 'All Files', extensions: ['*']}
```
Comment (sindresorhus) [inline main/menus.js]: We should not allow other file types.

File: main/menus.js [hunk 3]
```diff
@@ -40,6 +42,24 @@ const cogMenuTemplate = [
...
+        ],
+        properties: ['openFile']
+      }, filePaths => {
+        if (filePaths) {
+          filePaths.forEach(file => openEditorWindow(file));
```
Comment (sindresorhus) [inline main/menus.js]: Use a `for-of` loop instead. (Just my preference)

File: main/menus.js [hunk 4]
```diff
@@ -40,6 +42,24 @@ const cogMenuTemplate = [
...
+        }
+      });
+    }
+  },
```
Comment (sindresorhus) [inline main/menus.js]: I think we should make the order:

```
Open Video
Export History...
--------- (separator)
Preferences...
---------
```

File: main/menus.js [hunk 5]
```diff
@@ -40,6 +42,24 @@ const cogMenuTemplate = [
   {
     type: 'separator'
   },
+  {
+    label: 'Open Video…',
+    accelerator: 'Cmd+o',
+    click: () => {
+      closeAllCroppers();
```
Comment (sindresorhus) [inline main/menus.js]: Could use an empty newline below here.

File: main/common/constants.js [hunk 6]
```diff
@@ -0,0 +1,5 @@
+'use strict';
+
+const EXTENSIONS = ['mp4', 'mov', 'm4v'];
```
Comment (sindresorhus) [inline main/common/constants.js]: I would make the variable more descriptive. `const supportedVideoExtensions`.

**Comment by theshortcut** [comment]
updated menu item ordering:

image

**Comment by theshortcut** [comment]
@skllcrn thank you. I use gif captures pretty heavily during day to day work and recently replaced Giphy Capture with Kap v3 and its such an improvement.

**Comment by sindresorhus** [comment]
Thanks for this awesome contribution, @theshortcut :)

**Pull request merged**